apps/ex_nvr_web/lib/ex_nvr_web/controllers/api/event_controller.ex

```diff
@@ -0,0 +1,101 @@
+defmodule ExNVRWeb.API.EventController do
+  use ExNVRWeb, :controller
+
+  action_fallback ExNVRWeb.API.FallbackController
+
+  alias Plug.Conn
+  alias ExNVR.Model.Device
+  alias ExNVR.{Events, Devices}
+
+  @spec create(Conn.t(), map()) :: {:error, Ecto.Changeset.t()} | Conn.t()
+  def create(%Conn{body_params: body_params, query_params: query_params} = conn, _) do
+    device_id = Map.get(query_params, "device_id", "")
+    event_type = Map.get(query_params, "type", "")
+
+    with %Device{} = device <- Devices.get(device_id),
+          true <- event_type == "lpr",
+          {:ok, params, plate_picture, full_picture} <- format_params(device, body_params),
+          {:ok, _event} <- Events.create_lpr_event(params, plate_picture, full_picture)do
+      send_resp(conn, 201, "")
+    else
+      error -> {:error, error}
```

Comment:
It's better to have useful errors. Let's rewrite this to
```elixir
with {:ok, device} <- get_device(device_id),
       :ok <- check_event_type(event_type),
       ...
end
```
Add the new functions `get_device/1` and `check_event_type/1`. This function must return `{:error, :not_found}` in case of error, this way the fallback controller will send appropriate error message.